### PR TITLE
CrossTab UI

### PR DESCRIPTION
**Pull request by breville**
This introduces a kind of `CrossTab` UI control, shown when a label and one or more features are selected.  It has a row for each combination of feature values, percentages for the cells, and rudimentary heatmap colouring.

The idea is that students can spot features, and feature combinations, that have particularly strong predictive power over a label.

Screen Shot 2021-01-29 at 5 38 07 PM
Screen Shot 2021-01-29 at 5 38 32 PM
Screen Shot 2021-01-29 at 5 38 58 PM

### CODE REVIEW

File: src/redux.js [hunk 1]
```diff
@@ -898,3 +898,124 @@ export function getCurrentColumnIsSelectedLabel(state) {
...
+    for (let labelCount of Object.values(result.labelCounts)) {
+      totalCount += labelCount;
+    }
+    result.labelPercents = {};
+    for (let key of Object.keys(result.labelCounts)) {
+      result.labelPercents[key] = Math.round(
+        (result.labelCounts[key] / totalCount) * 100
+      );
+    }
+  }
```
Comment (Erin007) [inline src/redux.js:987]: Have you tried this with a big dataset? I have a general concern about performance in scenarios where we go through a whole column; not blocking for this, just curious.

File: src/redux.js [hunk 2]
```diff
@@ -898,3 +898,124 @@ export function getCurrentColumnIsSelectedLabel(state) {
...
+    for (let labelValue of labelValues) {
+      if (
+        !uniqueLabelValues.find(uniqueLabelValue => {
+          return uniqueLabelValue === labelValue;
+        })
+      ) {
+        uniqueLabelValues.push(labelValue);
+      }
+    }
+  }
```
Comment (Erin007) [inline src/redux.js:1004]: There's already a function`getUniqueOptions(state, state.currentColumn)`. Reusable here?

**Review by Erin007 — APPROVED** [review]
Thanks for pulling this together so quickly! Code LGTM with a couple questions for consideration. Curious to see how the Curriculum team feels about this direction.

File: src/redux.js (hunk 2 above)
Comment (breville) [reply src/redux.js]: Good call.  Will use that.

File: src/redux.js (hunk 1 above)
Comment (breville) [reply src/redux.js:987]: Good point.  I suspect we have a number of perf bottlenecks and we will need to understand how big the incoming datasets can get, but I haven't yet done that.

**Pull request merged**